### PR TITLE
it8xxx2: intc: Look for pending interrupt

### PR DESCRIPTION
**Pull request by Dino-Li**
The AIVECT (All Interrupt Vector Register) register provides us a quick
way to determine SOC interrupt number. But sometimes we get interrupt
number 0 from the register and caused assertion of unhandled interrupt.
To avoid the assertion, we look for all ISRs (Interrupt Status Register)
and IERs (Interrupt Enable Register) to find the highest priority pending
interrupt and return it to caller.

Signed-off-by: Dino Li <Dino.Li@ite.com.tw>

### CODE REVIEW

File: drivers/interrupt_controller/intc_ite_it8xxx2.c [hunk 1]
```diff
@@ -206,6 +206,25 @@ uint8_t get_irq(void *arg)
 	} while (intc_irq != IVECT);
 	/* determine interrupt number */
 	intc_irq -= IVECT_OFFSET_WITH_IRQ;
+	/*
+	 * Look for pending interrupt if there's interrupt number 0 from
+	 * the AIVECT register.
+	 */
+	if (!intc_irq) {
```
Comment (yperess) [inline drivers/interrupt_controller/intc_ite_it8xxx2.c:213]: nit: prefer explicit `== 0` as per coding guidelines

File: drivers/interrupt_controller/intc_ite_it8xxx2.c [hunk 2]
```diff
@@ -206,6 +206,25 @@ uint8_t get_irq(void *arg)
...
+		uint8_t int_pending;
+
+		for (int i = (IT8XXX2_IER_COUNT - 1); i >= 0; i--) {
+			int_pending = (*reg_status[i] & *reg_enable[i]);
+			if (int_pending) {
```
Comment (yperess) [inline drivers/interrupt_controller/intc_ite_it8xxx2.c:218]: Same here, prefer `!= 0`

File: drivers/interrupt_controller/intc_ite_it8xxx2.c [hunk 3]
```diff
@@ -206,6 +206,25 @@ uint8_t get_irq(void *arg)
...
+				intc_irq = (MAX_REGISR_IRQ_NUM * i) +
+						find_msb_set(int_pending) - 1;
+				printk("Pending interrupt found: %d\n",
+						intc_irq);
+				printk("CPU mepc: 0x%lx\n", csr_read(mepc));
```
Comment (yperess) [inline drivers/interrupt_controller/intc_ite_it8xxx2.c:223]: Can we use logging instead of `printk`? This seems more like a `LOG_DBG`

**Review by yperess — CHANGES_REQUESTED** [review]

File: drivers/interrupt_controller/intc_ite_it8xxx2.c (hunk 1 above)
Comment (Dino-Li) [reply drivers/interrupt_controller/intc_ite_it8xxx2.c]: Done

File: drivers/interrupt_controller/intc_ite_it8xxx2.c (hunk 2 above)
Comment (Dino-Li) [reply drivers/interrupt_controller/intc_ite_it8xxx2.c]: Done

File: drivers/interrupt_controller/intc_ite_it8xxx2.c (hunk 3 above)
Comment (Dino-Li) [reply drivers/interrupt_controller/intc_ite_it8xxx2.c]: Done

**Review by tari — APPROVED** [review]

**Review by yperess — APPROVED** [review]

**Review by GTLin08 — APPROVED** [review]

**Pull request merged**